ts/utils/biometrics.ts

```diff
@@ -2,52 +2,60 @@ import { Alert, Platform } from "react-native";
 import FingerprintScanner, {
   AuthenticateAndroid,
   AuthenticateIOS,
+  Biometrics,
   FingerprintScannerError
 } from "react-native-fingerprint-scanner";
 import { isDebugBiometricIdentificationEnabled } from "../config";
 import I18n from "../i18n";
 import { mixpanelTrack } from "../mixpanel";
-import { BiometrySimpleType } from "../sagas/startup/checkAcknowledgedFingerprintSaga";
+
+export type BiometricsErrorType =
+  // possibly working, but the string returned is undocumented
+  | "UNKNOWN"
+
+  // unspeakable horrors happened somewhere under the hood
+  | "UNAVAILABLE";
+
+export type BiometricsValidType =
+  // happy path
+  "BIOMETRICS" | "FACE_ID" | "TOUCH_ID";
+
+export type BiometricsType = BiometricsErrorType | BiometricsValidType;
 
 /**
  * Retrieve biometric settings from the base system. This function wraps the basic
  * method "isSensorAvailable" of react-native-fingerprint-scanner library and simplifies the possible returned values in
- * function of its usage:
- * * FaceID/TouchID/Biometrics: in case of biometric recognition supported.
- * * Throws an error in case the Biometric option is not enrolled or not available on the device
+ * function of its usage.
  *
  * More info about library can be found here: https://github.com/hieuvp/react-native-fingerprint-scanner
  */
-export function getFingerprintSettings(): Promise<BiometrySimpleType> {
-  return new Promise((resolve, _) => {
-    FingerprintScanner.isSensorAvailable()
-      .then(biometryType => {
-        switch (biometryType) {
-          case "Touch ID":
-            resolve("TOUCH_ID");
-            break;
-          case "Face ID":
-            resolve("FACE_ID");
-            break;
-          case "Biometrics":
-            resolve("BIOMETRICS");
-            break;
-          default:
-            resolve("UNAVAILABLE");
-            break;
-        }
-      })
-      .catch(e => {
-        void mixpanelTrack("BIOMETRIC_ERROR", { error: e });
-        resolve("UNAVAILABLE");
-      });
-  });
-}
+export const getBiometricsType = (): Promise<BiometricsType> =>
+  FingerprintScanner.isSensorAvailable()
+    .then((biometryType: Biometrics) => {
+      switch (biometryType) {
+        case "Touch ID":
+          return "TOUCH_ID";
+        case "Face ID":
+          return "FACE_ID";
+        case "Biometrics":
+          return "BIOMETRICS";
+        default:
+          return "UNKNOWN";
+      }
+    })
+    .catch(e => {
+      void mixpanelTrack("BIOMETRIC_ERROR", { error: e });
+      return "UNAVAILABLE";
+    });
+
+export const isBiometricsValidType = (
+  biometrics: BiometricsType
+): biometrics is BiometricsValidType => !biometrics.startsWith("UN");
```

Comment:
I guess there should be some cases where this type guard stops to work: i.e if we add a new case to the error types
```typescript
BiometricsErrorType =
  // new error
  | "SOME_FAILURE"
  // possibly working, but the string returned is undocumented
  | "UNKNOWN"
 
  // unspeakable horrors happened somewhere under the hood
  | "UNAVAILABLE";
```
or if we add a new case to the valid types
```typescript
export type BiometricsValidType =
  // happy path
  "BIOMETRICS" | "FACE_ID" | "TOUCH_ID" | "UNBELIEVABLE_BIOMETRIC";
```

`isBiometricsValidType("SOME_FAILURE") -> true`
`isBiometricsValidType("UNBELIEVABLE_BIOMETRIC") -> false`

An improvement could be to have the type guard loosely coupled to its type. Here my suggestion
```typescript
const biometricErrors = ["UNKNOWN", "UNAVAILABLE"] as const;
export type BiometricsErrorType = typeof biometricErrors[number];
[...]
export const isBiometricsValidType = (
  biometrics: BiometricsType
): biometrics is BiometricsValidType =>
  !biometricErrors.some(e => biometrics === e);
```
What do you think?

Reply:
I finally got a good use case for `as const` and I missed it 😮 😄 
Thank you so much for the suggestion! I was looking for something elegant to ensure type safety 🎉 

I've addressed this and the above suggestion in https://github.com/pagopa/io-app/pull/3393/commits/873cf593a5f4d35c9249bef9608e299dc35287ce